api/src/main/java/com/netflix/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -113,6 +117,11 @@ public R or(R leftResult, R rightResult) {
           return in(pred.ref(), pred.literal());
         case NOT_IN:
           return notIn(pred.ref(), pred.literal());
+        case STARTS_WITH:
+//          due to the fact that startWith function in Expressions accepts only string types casting is a must here.
+//          in  Unbound#bind function we added a check for that
+
+          return startsWith((BoundReference<String>) pred.ref(), pred.literal().to(pred.ref().type()));
```

Comment:
Why is this calling `to`? A `BoundReference` guarantees that the type of its literal is correct and has already been coerced during the binding process.